gossip/src/cluster_info.rs

```diff
@@ -2349,6 +2349,37 @@ pub struct NodeConfig {
     pub num_quic_endpoints: NonZeroUsize,
 }
 
+#[derive(Debug, Clone)]
+pub struct BindIpAddrs {
+    /// Primary IP address this node will bind to on startup
+    primary: IpAddr,
+    /// Secondary IP addresses
+    /// This node will bind to these addresses if the primary address is not available.
+    /// For some Solana protocols, this node will also bind to these addresses on startup
+    others: Vec<IpAddr>,
+}
+
+impl BindIpAddrs {
+    pub fn new(addrs: Vec<IpAddr>) -> Result<Self, String> {
+        if addrs.is_empty() {
+            return Err(
+                "BindIpAddrs requires at least one IP address (--bind-address)".to_string(),
+            );
+        }
+        let primary = addrs[0];
+        let others = addrs[1..].to_vec();
+        Ok(Self { primary, others })
+    }
+
+    pub fn primary(&self) -> IpAddr {
+        self.primary
+    }
+
+    pub fn all(&self) -> impl Iterator<Item = &IpAddr> {
```

Comment:
I would `impl Deref` into `&[IpAddr]` / `as_ref()->&[IpAddr]` (assuming we switch to Vec<IpAddr> representation). Then we can use itrators or whatever else downstream, and also get all the benefits of ExactSizeIterator if needed.